File: simple-stack-example-rx/src/main/java/com/zhuinden/stack_rx_example/FirstKey.java
package com.zhuinden.stack_rx_example;

import com.google.auto.value.AutoValue;

/**
 * Created by Owner on 2017. 02. 11..
 */

@AutoValue
public abstract class FirstKey extends Key {
    @Override
    public int layout() {
        return R.layout.path_first;
    }

    public static FirstKey create() {
        return new AutoValue_FirstKey();
    }
}

File: simple-stack-example-basic-kotlin/src/main/kotlin/com/zhuinden/simplestackdemoexample/common/FirstKey.kt
package com.zhuinden.simplestackdemoexample.common

import com.zhuinden.simplestack.navigator.ViewChangeHandler
import com.zhuinden.simplestack.navigator.changehandlers.NoOpViewChangeHandler
import com.zhuinden.simplestackdemoexample.R

import paperparcel.PaperParcel
import paperparcel.PaperParcelable

/**
 * Created by Owner on 2017. 01. 12..
 */

@Suppress("ProtectedInFinal")
@PaperParcel
data class FirstKey protected constructor(protected val tag: String = FirstKey::javaClass.name) //
    : PaperParcelable, Key {
    override fun layout(): Int {
        return R.layout.path_first
    }

    override fun viewChangeHandler(): ViewChangeHandler {
        return NoOpViewChangeHandler()
    }

    companion object {
        @JvmField val CREATOR = PaperParcelFirstKey.CREATOR

        fun create(): FirstKey {
            return FirstKey()
        }
    }
}
